v19.2/vectorized-execution.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Vectorized Query Execution
+summary: The CockroachDB vectorized SQL query execution engine processes query plans using a column-oriented model to improve performance.
+toc: true
+---
+
+ <span class="version-tag">New in v19.2:</span> CockroachDB supports column-oriented ("vectorized") query execution on all [supported SQL operations](https://www.cockroachlabs.com/docs/stable/sql-feature-support.html).
+
+ Many SQL databases, including MySQL, PostgreSQL, and all versions of CockroachDB released prior to 19.2, execute [query plans](https://en.wikipedia.org/wiki/Query_plan) one row of table data at a time. The CockroachDB vectorized execution engine dramatically improves performance over row-oriented execution by processing each component of a query plan on type-specific batches of column data. Column-oriented execution models contrast with row-oriented execution models, which can offer good performance for [online transaction processing (OLTP)](https://en.wikipedia.org/wiki/Online_transaction_processing) queries, but suboptimal performance for [online analytical processing (OLAP)](https://en.wikipedia.org/wiki/Online_analytical_processing) queries.
+
+
+## Configuring vectorized execution
+
+By default, vectorized execution is enabled in CockroachDB for all supported "streaming" SQL operations (i.e. operations that do not require [memory buffering](https://en.wikipedia.org/wiki/Data_buffer)). Non-streaming operations include global [sorts](query-order.html), [window functions](window-functions.html), [hash joins](joins.html#hash-joins), and [merge joins](joins.html#merge-joins) on non-unique columns. Merge joins on columns that are guaranteed to have one row per value, also known as "key columns", are considered streaming operations.
+
+You can turn vectorized execution on or off for all operations with the `sql.defaults.vectorize` [cluster setting](cluster-settings.html) or the `vectorize` [session variable](set-vars.html). The following options are supported for both the cluster setting and session variable:
+
+Option | Description
+----------|------------
+`auto` | Instructs CockroachDB to use the column-oriented vectorized SQL execution engine on streaming operations. By default, `vectorize=auto`.
+`experimental_on` | Turns vectorization on for all supported operations. We do not recommend using this setting in production environments, as it can lead to memory issues. See [Known Limitations](vectorized-execution.html#known-limitations) for more information.
```

Comment:
This "for all supported operations" language (here and elsewhere) is just highly confusing to me.  If something will lead to memory issues or a node crashing, then it's not really supported is it?  My mental model is this: If I were reading docs about something non-CRDB, say the Linux kernel, and it said "this module works with all supported kernel configurations" but then also said "ps if you enable this module, kernel could OOM your processes or crash your machine altogether" I would be *extremely* wary of whether it was in fact "supported".  It would really mess with my trust in the rest of the docs to even have that language in there.  Do you know what I mean?  Am I just crazy?  I think this is a larger conversation cc @jordanlewis @awoods187